File: DONGUK/프로그래머스_WEEK 4/체육복.py
def solution(n, lost, reserve):
    answer = 0
    
    new_lost = []
    for num in lost:
        if num in reserve:
            reserve.remove(num)
        else:
            new_lost.append(num)
    
    new_lost.sort()
    reserve.sort()
    
    answer = n - len(new_lost)
    
    while reserve:
        top = reserve.pop(0)
    
        if top-1 in new_lost:
            new_lost.remove(top-1)
            answer += 1
        
        elif top+1 in new_lost:
            new_lost.remove(top+1)
            answer += 1
    
    return answer

File: JINSEOP/프로그래머스_WEEK4/체육복.cpp
#include <string>
#include <vector>
using namespace std;

int stud[31];
void init_arr(vector<int> lost, vector<int> reserve) {
	for (int i : reserve) stud[i]++;
	for (int i : lost) stud[i]--;
}

int solution(int n, vector<int> lost, vector<int> reserve) {
	int answer = n;
	init_arr(lost, reserve);

	for (int i = 1; i <= n; i++) {
		if (stud[i] == -1) {
			if (stud[i - 1] == 1) {
				stud[i]++; stud[i - 1]--;
			}
			else if (stud[i + 1] == 1) {
				stud[i]++; stud[i + 1]--;
			}
		}
	}

	for (int i = 1; i <= n; i++)
		if (stud[i] == -1) answer--;
	return answer;
}
